aepp/src/utils/walletSearch.js

```diff
@@ -1,75 +1,97 @@
-    const scannerConnection = await BrowserWindowMessageConnection({
-      connectionInfo: { id: 'spy' },
-    });
-    const detector = await Detector({ connection: scannerConnection });
-    const handleWallets = async function ({ wallets, newWallet }) {
-      detector.stopScan();
-      const connected = await this.client.connectToWallet(await newWallet.getConnection());
-      this.client.selectNode(connected.networkId); // connected.networkId needs to be defined as node in RpcAepp
-      await this.client.subscribeAddress('subscribe', 'current');
-      aeternity.client = this.client;
-      aeternity.static = false;
-      await aeternity.initProvider();
-      successCallback();
-    };
-
-    detector.scan(handleWallets.bind(this));
-  },
-  async init (successCallback) {
-    // Open iframe with Wallet if run in top window
-    // window !== window.parent || await this.getReverseWindow();
-
-    this.client = await RpcAepp({
-      name: 'AEPP',
-      nodes: [
-        {name: 'ae_mainnet', instance: await Node({url: config.nodeUrl.ae_mainnet})},
-        {name: 'ae_uat', instance: await Node({url: config.nodeUrl.ae_uat})}
-      ],
-      compilerUrl: config.compilerUrl,
-      onNetworkChange (params) {
-        this.selectNode(params.networkId); // params.networkId needs to be defined as node in RpcAepp
-        aeternity.initProvider();
-      },
-      onAddressChange(addresses) {
-        if(!addresses.current[aeternity.address])
+  async initWalletOrFallbackStatic() {
+    wallet.walletStatus = 'connecting'
+
+    try {
+      // try to connect to Superhero Wallet
+      aeternity.client = new AeSdkAepp({
+        name: 'AEPP',
+        nodes,
+        onNetworkChange: async (network) => {
+          console.info('onNetworkChange:', network, aeternity.networkId !== network.networkId)
+          await this.aeConnectToNode(network.networkId)
+        }, onAddressChange: async (addresses) => {
+          console.info('onAddressChange:', addresses)
+          await this.aeConnectToNode(aeternity.networkId)
           EventBus.$emit('addressChange');
+        },
+      })
+
+      await this.scanForWallets()
+
+      if (wallet.walletStatus === 'fallback_static') {
+        aeternity.static = true;
+
+        aeternity.client = new AeSdk({nodes})
+
+        await this.aeConnectToNode(config.defaultNetworkId)
       }
-    });
+    } catch (error) {
+      console.error('initWallet error:', error)
+      return false
+    }
+
+    return true
+  },
+
+  async scanForWallets() {
+    wallet.walletStatus = 'scanning'
+
+    const detectedWallet = await Promise.race([new Promise((resolve) => {
+      const handleWallets = async () => {
+        stopScan()
+        resolve(true);
+      }
+
+      const scannerConnection = new BrowserWindowMessageConnection()
+      const stopScan = walletDetector(scannerConnection, handleWallets)
+    }), new Promise((resolve) => setTimeout(() => {
+      resolve(false);
+    }, 1000))]);
```

Comment:
maybe longer timeout? I think we initially had 8 seconds

Reply:
this seems to work fine with the updated connection logic (copied from governance aepp)